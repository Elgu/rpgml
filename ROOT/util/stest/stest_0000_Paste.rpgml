
{
  int[] dest = [ 0 to 9 ];
  int[] in = [ 13, 12, 11 ];

  print( .util.paste( in, dest, -3 ) );
  print( "\n" );

  print( .util.paste( in, dest, 10 ) );
  print( "\n" );

  print( .util.paste( in, dest, 0 ) );
  print( "\n" );

  print( .util.paste( in, dest, -1 ) );
  print( "\n" );

  print( .util.paste( in, dest, 8 ) );
  print( "\n" );
}

{
  int[,] in =
    [ 15, 14;
      13, 12;
      11, 10 ];
  int[,] dest = 
    [ 0 to 4;
      5 to 9;
      10 to 14;
      15 to 19 ];

  print( util.paste( in, dest, -2, -3 ) );
  print( "\n" );

  print( util.paste( in, dest, 0, 0 ) );
  print( "\n" );

  print( util.paste( in, dest, -1, -2 ) );
  print( "\n" );

  print( util.paste( in, dest, 4, 3 ) );
  print( "\n" );
}

exit();

