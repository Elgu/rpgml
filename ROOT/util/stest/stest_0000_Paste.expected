[ 0, 1, 2, 3, 4, 5, 6, 7, 8, 9 ]
[ 0, 1, 2, 3, 4, 5, 6, 7, 8, 9 ]
[ 13, 12, 11, 3, 4, 5, 6, 7, 8, 9 ]
[ 12, 11, 2, 3, 4, 5, 6, 7, 8, 9 ]
[ 0, 1, 2, 3, 4, 5, 6, 7, 13, 12 ]
[ 0, 1, 2, 3, 4; 5, 6, 7, 8, 9; 10, 11, 12, 13, 14; 15, 16, 17, 18, 19 ]
[ 15, 14, 2, 3, 4; 13, 12, 7, 8, 9; 11, 10, 12, 13, 14; 15, 16, 17, 18, 19 ]
[ 10, 1, 2, 3, 4; 5, 6, 7, 8, 9; 10, 11, 12, 13, 14; 15, 16, 17, 18, 19 ]
[ 0, 1, 2, 3, 4; 5, 6, 7, 8, 9; 10, 11, 12, 13, 14; 15, 16, 17, 18, 15 ]
